### DISCUSSION TITLE
No tldr entry for ***

### DISCUSSION THREAD

**Discussion by teamtee** [tldr-pages/tldr#9768] [Q&A] [answered]
What I do is below:
(1)sudo apt install tldr 
(2)sudo tldr --update
But when I write down this:
%tldr tar
it shows that:
No tldr entry for tar

**Comment by kbdharun** [comment, marked as answer]
We don't have an official client for Ubuntu directly in the repos. You can install the Python client using `snap install tldr`. Also, you don't need to use `sudo` with `tldr --update` or `tldr -u` (as they are used for updating the pages locally and don't require elevated access).

**Reply by CleanMachine1** [reply]
Most likely there is no page due to the sudo being used when updating, the client (haskell I think), probably uses the $USER variable to choose a location for the tldr pages and using sudo will make it save to roots home.

**Reply by teamtee** [reply]
Thank you for your earnest answer

**Reply by levid0s** [reply]
I just stumbled upon this project and ran into the same issue.

It would be nice if `tldr` could detect that there are 0 pages downloaded and suggest running `tldr -u`

**Comment by kbdharun** [comment]
> I just stumbled upon this project and ran into the same issue.
> 
> It would be nice if `tldr` could detect that there are 0 pages downloaded and suggest running `tldr -u`

Hi, welcome to tldr @levid0s. Can you share which client you are using? 

Most of our official clients fetch pages by default when there isn't a local index or it tells you to run `tldr --update` or `tldr -u`.

**Reply by levid0s** [reply]
On a vanilla Ubuntu, a new user would intuitively do the following;
But `apt` installs quite an old version, so there might not be much you can do about it.

```
lev@test-nginx-qbbw:~$ cat /etc/issue
Ubuntu 22.04.2 LTS \n \l
lev@test-nginx-qbbw:~$ tldr
Command 'tldr' not found, but can be installed with:
snap install tldr     # version 3.2.0, or
apt  install tldr     # version 0.6.4-1build6.3
apt  install tldr-py  # version 0.7.0-4
See 'snap info tldr' for additional versions.
lev@test-nginx-qbbw:~$ sudo apt install -y tldr
....
lev@test-nginx-qbbw:~$ tldr -v
0.6.4
lev@test-nginx-qbbw:~$ tldr tr
No tldr entry for tr
```

**Reply by kbdharun** [reply]
The package in Ubuntu repos is from the Debian package for `tldr-hs` (an unofficial client), the reason for this error is that the latest working version isn't backported from the interim release repositories (i.e 23.04, etc) to the LTS one (i.e 22.04 LTS, etc). I have [contacted the Debian package maintainers](https://github.com/tldr-pages/tldr/issues/10198) for `tldr` suggesting renaming it to `tldr-hs` and replacing the current one with our Official Python client instead and some progress is being made.

For now, I would suggest installing our Official Python client using `snap install tldr` or `pip3 install tldr` [which is the easiest method]. If you have NPM (NodeJS) or Homebrew installed, then you use our official Node (`npm install -g tldr`) or C client (`brew install tldr`) too.

Edit. An update on the package,  check out #10533

**Comment by EagleEyez20** [comment]
I know this thread is old but in case others find it like I did here is the fix.

$ tldr -u

![Screenshot from 2024-09-29 16-31-07](https://github.com/user-attachments/assets/028f5889-ce07-4d32-8117-0172e2879a0e)

**Comment by mrx23dot** [comment]
How should I know that I need to execute
tldr --update
?
it's not in the readme.md

**Reply by kbdharun** [reply]
Agreed, we should probably document common commands in a usage section (comprising of the `tldr` page that you get when typing `tldr tldr`) in the README here and on the Wiki's home page. Ideally, this command is autorun when a page isn't found in the cache (this is how our official clients work) or else the command would be printed in the client's terminal to be executed by the user.
